test/unit/test_intrin.cu

```diff
@@ -46,34 +45,42 @@ namespace kernels {
 TEST(gpu_intrinsics, cuda_atomic_add) {
     int expected = (128*129)/2;
 
-    auto f = arb::gpu::make_managed_ptr<float>(0.f);
-    kernels::test_atomic_add<<<1, 128>>>(f.get());
+    arb::memory::device_vector<float> f(1);
+    f[0] = 0.f;
+
+    kernels::test_atomic_add<<<1, 128>>>(f.data());
     cudaDeviceSynchronize();
```

Comment:
we don't need the `cudaDeviceSynchronize` calls now that we don't use managed memory (it was required to stop the host from accessing the memory before the kernel had finished modifying the memory).
The access below (i.e. `f[0]` on line 54) uses `cudaMemcpy` under the hood, which will be synchronized with the kernel launch.